src/journal/journald-server.c

```diff
@@ -2172,9 +2277,21 @@ int server_init(Server *s) {
         server_cache_boot_id(s);
         server_cache_machine_id(s);
 
-        s->runtime_storage.path = path_join("/run/log/journal", SERVER_MACHINE_ID(s));
-        s->system_storage.path  = path_join("/var/log/journal", SERVER_MACHINE_ID(s));
-        if (!s->runtime_storage.path || !s->system_storage.path)
+        if (s->namespace)
+                s->runtime_storage.path = strjoin("/run/log/journal/", SERVER_MACHINE_ID(s), ".", s->namespace);
+        else
+                s->runtime_storage.path = strjoin("/run/log/journal/", SERVER_MACHINE_ID(s));
```

Comment:
Yes, I initially wrote the code like that and had them as subdirs. But the client side was became really ugly, since suddenly we'd not just watch two levels of dirs withe journal files in the inner but three levels of dirs with journal files in the inner two if you follow what I mean.

(i.e. think `journalctl --merge` where we watch /var/log/journal/ as dir + /var/log/journal/*/ as dir and /var/log/journal/*/*.journal as files, currently)

So this was just ugly I thought, but not convincing alone to not nest it. But the thing that kind broke the camels back for me is that vacuuming (and kinda also rotation, syncing, …) are something we defined so far as being "per-machine-directory", i.e. we'd look into all files inside the /var/log/journal/$MACHINE_ID/ and remove stuff until our constraints are valid again. So this so far was a "pruning of the leaves" of the tree. But if we'd nest namepsace stuff we'd not "prune the leaves" anymore but prune something "in the middle" in many cases, as the assumption would be that vaccuuming the host journald would not mean that we'd vaccuum the namespace dirs too, right?

I figure conceptually I think nesting the namespace dirs in the main journal dirs would mean that all operations applied to the main journal dirs should also affect the namespace dirs. But I think most likely this is not what we want: namespaces after all are about isolating parts of the system from the rest, and treating things as separate as we can, hence the operations should be separate and thus the dirs too.

Does that make any sense?